addons/stock/models/stock_move.py

```diff
@@ -2159,14 +2159,34 @@ def _trigger_assign(self):
                                                          order='reservation_date, priority desc, date asc, id asc')
         moves_to_reserve._action_assign()
 
+    def _rollup_move_dests_fetch(self):
+        seen = set(self.ids)
+        self.fetch(['move_dest_ids'])
+        move_dest_ids = set(self.move_dest_ids.ids)
+        while not move_dest_ids.issubset(seen):
+            seen |= move_dest_ids
+            to_visit = self.browse(move_dest_ids)
+            to_visit.fetch(['move_dest_ids'])
+            move_dest_ids = set(to_visit.move_dest_ids.ids)
+
+    def _rollup_move_origs_fetch(self):
+        seen = set(self.ids)
+        self.fetch(['move_orig_ids'])
+        move_orig_ids = set(self.move_orig_ids.ids)
+        while not move_orig_ids.issubset(seen):
+            seen |= move_orig_ids
+            to_visit = self.browse(move_orig_ids)
+            to_visit.fetch(['move_orig_ids'])
+            move_orig_ids = set(to_visit.move_orig_ids.ids)
+
     def _rollup_move_dests(self, seen=False):
         if not seen:
             seen = OrderedSet()
         unseen = OrderedSet(self.ids) - seen
         if not unseen:
             return seen
         seen.update(unseen)
-        self.filtered(lambda m: m.id in unseen).move_dest_ids._rollup_move_dests(seen)
+        self.browse(m for m in self.ids if m in unseen).move_dest_ids._rollup_move_dests(seen)
```

Comment:
Is the difference with `filtered` really noticeable ? `filtered` does have an overhead but for such a trivial condition it should be negligible no ?

Reply:
I think there was some noticeable difference while I was testing but I wasn't sure why. I may test this again just to be sure and can try to see why exactly

Reply:
Ok my mistake there actually is no difference here 😅 I will remove these

Reply:
Removed in amend